src/main/java/org/jenkinsci/plugins/workflow/cps/CpsGroovyShell.java

```diff
@@ -110,7 +110,7 @@ public void prepareScript(Script script) {
     public Script parse(GroovyCodeSource codeSource) throws CompilationFailedException {
         Script s = doParse(codeSource);
         if (execution!=null)
-            execution.loadedScripts.put(s.getClass().getName(), codeSource.getScriptText());
+            execution.loadedScripts.put(s.getClass().getSimpleName(), codeSource.getScriptText());
```

Comment:
Um, if the issue is the fact that the key is a String and is duplicated, won't keying by simple name make it worse?  As in, rather than key collisions on *all* components (package and class simple name) it'll happen if just the class's base name matches?   Or am I missing something here about the impact of the keying?  For reference: https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav